src/main/java/functions/CheckFunction.kt

```diff
@@ -0,0 +1,22 @@
+package functions
+
+import tree.TargetTreeNode
+
+fun checkFunction(expression: VisitorFunction): VisitorFunction = { targetTreeNode ->
+    var currentNode = targetTreeNode
+    var listAnswer = listOf<List<TargetTreeNode>>()
+    val allAnswers = mutableListOf<VisitorFunctionResponse>()
+    while (currentNode != null) {
+        if (listAnswer.find { it == currentNode } == null) {
```

Comment:
Does it ever ignore the `if` ? Taking into account that `listAnswer` is a `List<List<TargetTreeNode>>`, so `it` is `List<TargetTreeNode>` and we are comparing it with a `TargetTreeNode`. 

I'm not sure if `==` can be applied between lists and single values in kotlin, but in case it is a mistake here could be a solution:

```suggestion
        if (listAnswer.find { it.any { it == currentNode } } == null) {
```

Reply:
Yes, check the code update that is an error I fixed in the changes, thanks for the review.

Reply:
I can't see an updated version with that error fixed. Have you already pushed it?